src/js/components/DateInput/DateInput.js

```diff
@@ -152,9 +153,21 @@ const DateInput = forwardRef(
                 // clicking an edge date removes it
                 else if (range) normalizedValue = [nextValue, nextValue];
                 else normalizedValue = nextValue;
+                let nextNormalize = normalize;
+                if (timestamp === undefined) {
+                  nextNormalize = false;
+                  setNormalize(nextNormalize);
```

Comment:
Do we need to do something similar in the `onChange` for MaskedInput, when the user enters the first date by typing it?

Reply:
Hmm good thought. let me see.